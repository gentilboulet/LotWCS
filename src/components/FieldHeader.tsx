import * as React from 'react';

export interface IFieldHeader { label: string | JSX.Element; }

class FieldHeader extends React.Component<IFieldHeader, {}> {
  public render(): JSX.Element {
    return (
      <h4>
        <span className="align-text-middle" >{this.props.label}</span>
      </h4>
    );
  }
}

export default FieldHeader;
