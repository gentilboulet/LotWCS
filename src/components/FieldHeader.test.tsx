import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import FieldHeader from './FieldHeader';

describe('Testing the FieldHeader component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    const element = <FieldHeader label="something" />;
    ReactDOM.render(element, div);
    expect( renderer.create(element).toJSON() ).toMatchSnapshot();
  });
});
