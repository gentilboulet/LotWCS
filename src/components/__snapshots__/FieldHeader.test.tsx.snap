// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the FieldHeader component should render without crashing 1`] = `
<h4>
  <span
    className="align-text-middle"
  >
    something
  </span>
</h4>
`;
